src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -31,11 +31,18 @@
 #include "utilities/macros.hpp"
 
 const char* VM_Version::_uarch = "";
+const char* VM_Version::_vm_mode = "";
 uint32_t VM_Version::_initial_vector_length = 0;
 
 void VM_Version::initialize() {
   get_os_cpu_info();
 
+  // check if satp.mode is supported, currently supports up to SV48(rv64)/SV32(rv32)
+  VM_MODE mode = get_satp_mode();
+  if (mode > RISCV64_ONLY(VM_SV48) RISCV32_ONLY(VM_SV32)) {
+    vm_exit_during_initialization(err_msg("Unsupported satp mode: %d", mode));
```

Comment:
I would map that number to a string. For example 10 to “sv48” so it is more googlable and searchable.

Reply:
Thanks for the suggestion, actually we can just print `_vm_mode` since it already has mmu string.